src/main/java/mate.academy/lib/Injector.java

```diff
@@ -1,9 +1,73 @@
 package mate.academy.lib;
 
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+import java.util.HashMap;
+import java.util.Map;
+import mate.academy.service.FileReaderService;
+import mate.academy.service.ProductParser;
+import mate.academy.service.ProductService;
+import mate.academy.service.impl.FileReaderServiceImpl;
+import mate.academy.service.impl.Inject;
+import mate.academy.service.impl.ProductParserImpl;
+import mate.academy.service.impl.ProductServiceImpl;
+
 public class Injector {
     private static final Injector injector = new Injector();
+    private Map<Class<?>, Object> instances = new HashMap<>();
 
     public static Injector getInjector() {
         return injector;
     }
+
+    public Object getInstance(Class<?> interfaceClass) {
+        Object classImplInstance = null;
+        Class<?> clazz = findImplementation(interfaceClass);
+        Field[] declaredField = interfaceClass.getDeclaredFields();
+        for (Field field : declaredField) {
+            if (field.isAnnotationPresent(Inject.class)) {
+                Object fieldInstance = getInstance(field.getType());
+                classImplInstance = createNewInstance(clazz);
+                field.setAccessible(true);
+                try {
+                    field.set(classImplInstance, fieldInstance);
+                } catch (IllegalAccessException e) {
+                    throw new RuntimeException("Can't initialize field value. Class; "
+                            + clazz.getName()
+                    + "Field:|" + field.getName());
+                }
+            }
+        }
+        if (classImplInstance == null) {
+            classImplInstance = createNewInstance(clazz);
+        }
+        return classImplInstance;
+    }
+
+    private Object createNewInstance(Class<?> clazz) {
+        if (instances.containsKey(clazz)) {
+            return instances.get(clazz);
+        }
+        try {
+            Constructor<?> constructor = clazz.getConstructor();
+            Object instance = constructor.newInstance();
+            instances.put(clazz, instance);
+            return instance;
+        } catch (NoSuchMethodException | InstantiationException
+                | IllegalAccessException | InvocationTargetException e) {
+            throw new RuntimeException("Can't create new instatnce of " + clazz.getName());
+        }
+    }
+
+    private Class<?> findImplementation(Class<?> interfaceClass) {
+        Map<Class<?>, Class<?>> interfaceImpl = new HashMap<>();
+        interfaceImpl.put(FileReaderService.class, FileReaderServiceImpl.class);
+        interfaceImpl.put(ProductService.class, ProductServiceImpl.class);
+        interfaceImpl.put(ProductParser.class, ProductParserImpl.class);
+        if (interfaceClass.isInterface()) {
+            return interfaceImpl.get(interfaceClass);
+        }
+        return interfaceClass;
+    }
```

Comment:
Maybe it is better to add 'if' statement, that checks if the class is marked with @Component annotation. Otherwise:  RuntimeExeption, telling that "This class is not marked with 'Component' annotation" ?